rofl-containers/src/proxy/mod.rs

```diff
@@ -0,0 +1,161 @@
+    let proxy_label: ProxyLabel = match la.labels.get(LABEL_PROXY) {
+        None => return Ok(()),
+        Some(value) if value.is_empty() => return Ok(()),
+        Some(value) => {
+            let proxy_label = BASE64_STANDARD.decode(value)?;
+            let proxy_label = kms
+                .open_secret(&services::kms::OpenSecretRequest {
+                    name: "",
+                    value: &proxy_label,
+                    context: Some(PROXY_LABEL_ENCRYPTION_CONTEXT),
+                })
+                .await?
+                .value;
+            cbor::from_slice(&proxy_label)?
+        }
+    };
+
+    slog::info!(logger, "proxy configuration is available, starting proxy");
+    tokio::spawn(async move {
+        if let Err(err) = run(proxy_label).await {
+            slog::error!(logger, "failed to start proxy for containers"; "err" => ?err);
+        }
+    });
+
+    Ok(())
+}
+
+async fn run(proxy_label: ProxyLabel) -> Result<()> {
+    let logger = get_logger("proxy");
+
+    // We only need the IP address part of the IP/CIDR format.
+    let listen_address = proxy_label
+        .wireguard
+        .address
+        .split("/")
+        .next()
+        .ok_or(anyhow!("bad proxy listen address"))?
+        .to_string();
+
+    let hub_address = proxy_label
+        .wireguard
+        .hub_address
+        .split("/")
+        .next()
+        .ok_or(anyhow!("bad hub address"))?;
+
+    // Setup firewall.
+    slog::info!(logger, "setting up firewall");
+    let mut firewall = Firewall::new();
+    firewall
+        .add_wireguard(
+            WG_INTERFACE_NAME,
+            hub_address,
+            &listen_address,
+            PROXY_HTTPS_LISTEN_PORT,
+        )
+        .context("failed to add proxy firewall rules")?;
+    firewall.start().context("failed to start firewall")?;
+
+    // Setup wireguard tunnel.
+    slog::info!(logger, "setting up wireguard");
+    let wg_cfg = wireguard::ClientConfig {
+        listen_port: Some(wireguard::WG_DEFAULT_LISTEN_PORT),
+        ..proxy_label.wireguard.clone()
+    };
+    let _wireguard = wireguard::Client::new(wg_cfg).context("failed to start wireguard client")?;
+
+    // Setup HTTPS proxy.
+    slog::info!(logger, "setting up application proxy");
+    let mut http = http::Proxy::new(http::Config {
+        mode: http::Mode::TerminateTls,
+        listen_address: listen_address.clone(),
+        listen_port: PROXY_HTTPS_LISTEN_PORT,
+        ..Default::default()
+    })
+    .context("failed to start https proxy")?;
+    http.start();
+
+    let data = fs::read_to_string(COMPOSE_FILE_PATH).context("failed to load compose file")?;
+    let compose = ParsedCompose::parse(&data)?;
+    for mapping in compose.port_mappings {
+        if mapping.port.protocol != "tcp" {
+            continue;
+        }
+
+        let name = format!("p{}.{}", mapping.port.host_port, proxy_label.http_host);
+
+        slog::info!(logger, "adding mapping for port";
```

Comment:
I think we also discussed that it might make sense to support disabling proxying for a specific port. Do you plan to include that in this PR, or leave it for a follow-up?